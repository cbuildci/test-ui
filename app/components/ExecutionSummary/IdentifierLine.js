/**
 * IdentifierLine
 */

import React from 'react';
import PropTypes from 'prop-types';

function IdentifierLine({
    githubHost,
    owner,
    repo,
    commit,
    executionNum,
}) {
    const urlPrefix = githubHost
        ? `https://${githubHost}/`
        : '/app/repo/';

    return (
        <span className="d-inline-flex flex-wrap">
            <span className="mr-2">
                <i className="fab fa-github fa-fw text-muted mr-1"/>
                <a href={`${urlPrefix}${owner}/${repo}`}>{owner}/{repo}</a>
            </span>

            <span className="mr-2">
                <i className="fas fa-code-branch fa-fw text-muted mr-1"/>
                <a href={`${urlPrefix}${owner}/${repo}/commit/${commit}`}>{commit.substr(0, 10)}</a>
            </span>

            <span className="mr-2">
                <i className="fas fa-shipping-fast fa-fw text-muted mr-1"/>
                #{executionNum}
            </span>
        </span>
    );
}

IdentifierLine.propTypes = {
    githubHost: PropTypes.string,
    owner: PropTypes.string.isRequired,
    repo: PropTypes.string.isRequired,
    commit: PropTypes.string.isRequired,
    executionNum: PropTypes.number.isRequired,
};

IdentifierLine.defaultProps = {
    githubHost: null,
};

export default IdentifierLine;
