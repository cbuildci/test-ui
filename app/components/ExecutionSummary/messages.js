/*
 * ExecutionSummary Messages
 */

import { defineMessages } from 'react-intl';

const baseStarted = 'Started {relativeCreateTimeHtml}';

export default defineMessages({
    startedWithoutEvent: {
        id: 'app.components.ExecutionSummary.startedWithoutEvent',
        defaultMessage: baseStarted,
    },
    startedWithUnknownEvent: {
        id: 'app.components.ExecutionSummary.startedWithUnknownEvent',
        defaultMessage: `${baseStarted} after event "{event}"`,
    },
    startedForPullRequest: {
        id: 'app.components.ExecutionSummary.startedForPullRequest',
        defaultMessage: `${baseStarted} after {action, select,
            opened {{actionHtml}}
            synchronize {{actionHtml} to}
            other {{actionHtml}}
        }
        {pullRequestLinkHtml}`,
    },
    startedForPullRequest_action: {
        id: 'app.components.ExecutionSummary.startedForPullRequest_action',
        defaultMessage: `{action, select,
            opened {opening}
            synchronize {push}
            other {action}
        }`,
    },
    startedForPullRequest_link: {
        id: 'app.components.ExecutionSummary.startedForPullRequest_link',
        defaultMessage: 'pull request #{pullRequestNumber}',
    },
    startedByUserAction: {
        id: 'app.components.ExecutionSummary.startedByUserAction',
        defaultMessage: `${baseStarted} {identifier, select,
            rerun {as {identifierHtml}}
            other {for action {identifierHtml}}
        } initiated by {usernameHtml}`,
    },
    startedByUserAction_identifier: {
        id: 'app.components.ExecutionSummary.startedByUserAction_identifier',
        defaultMessage: `{identifier, select,
            rerun {re-run}
            other {{identifier}}
        }`,
    },
    userStop: {
        id: 'app.components.ExecutionSummary.userStop',
        defaultMessage: 'Stop requested after {durationHtml} by {userHtml}',
    },
    conclusionMessage: {
        id: 'app.components.ExecutionSummary.conclusionMessage',
        defaultMessage: `{hasStop, select, 
            yes {Stop requested after {stopDurationHtml} by {stopUserHtml} {hasConcluded, select,
                yes {and took {conclusionDurationHtml} to stop}
                no {}
            }}
            no {Concluded after {conclusionDurationHtml}}
        }`,
    },
});
