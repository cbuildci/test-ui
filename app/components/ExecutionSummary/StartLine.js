/**
 * StartLine
 */

import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage, FormattedRelative } from 'react-intl';
import messages from './messages';
import Time from 'components/Time';

function StartLine({
    githubHost,
    owner,
    repo,
    createTime,
    event,
}) {
    let message = messages.startedWithoutEvent;

    const values = {
        relativeCreateTimeHtml: (
            <strong>
                <Time date={createTime}>
                    <FormattedRelative
                        value={createTime}
                        updateInterval={5000}
                    />
                </Time>
            </strong>
        ),
    };

    if (event) {
        message = messages.startedWithUnknownEvent;
        values.event = event.event;

        if (event.event === 'pull_request') {
            message = messages.startedForPullRequest;

            values.action = event.action;
            values.actionHtml = (
                <strong>
                    <FormattedMessage
                        {...messages.startedForPullRequest_action}
                        values={{ action: event.action }}
                    />
                </strong>
            );

            values.pullRequestLinkHtml = (
                <a href={`https://${githubHost}/${owner}/${repo}/pull/${event.pull_request.number}`}>
                    <FormattedMessage
                        {...messages.startedForPullRequest_link}
                        values={{ pullRequestNumber: event.pull_request.number }}
                    />
                </a>
            );
        }
        else if (event.event === 'check_run'
            && event.action === 'requested_action'
            && event.requested_action.identifier === 'rerun') {

            message = messages.startedByUserAction;

            values.username = event.sender.login;
            values.usernameHtml = (
                <a href={`https://${githubHost}/${event.sender.login}`}>@{event.sender.login}</a>
            );

            values.identifier = event.requested_action.identifier;
            values.identifierHtml = (
                <strong>
                    <FormattedMessage
                        {...messages.startedByUserAction_identifier}
                        values={{ identifier: event.requested_action.identifier }}
                    />
                </strong>
            );
        }
    }

    return (
        <FormattedMessage
            {...message}
            values={values}
        />
    );
}

StartLine.propTypes = {
    githubHost: PropTypes.string.isRequired,
    owner: PropTypes.string.isRequired,
    repo: PropTypes.string.isRequired,
    createTime: PropTypes.oneOfType([
        PropTypes.number,
        PropTypes.string,
        PropTypes.instanceOf(Date),
    ]).isRequired,
    conclusionTime: PropTypes.oneOfType([
        PropTypes.number,
        PropTypes.string,
        PropTypes.instanceOf(Date),
    ]),
    event: PropTypes.object,
};

StartLine.defaultProps = {
    conclusionTime: null,
    event: null,
};

export default StartLine;
