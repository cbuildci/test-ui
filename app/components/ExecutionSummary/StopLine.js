/**
 * StopLine
 */

import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import messages from './messages';
import TimeDuration from 'components/TimeDuration';
import Time from 'components/Time';
import { getDateTime } from '../../utils/date';

function StopLine({
    githubHost,
    stopUser,
    createTime,
    conclusionTime,
    stopRequestTime,
}) {
    const values = {
        hasConcluded: 'no',
        hasStop: 'no',
    };

    createTime = Math.round(getDateTime(createTime) / 1000);

    if (stopRequestTime != null) {
        stopRequestTime = Math.round(getDateTime(stopRequestTime) / 1000);
        values.hasStop = 'yes';

        values.stopDurationHtml = (
            <strong>
                <Time date={stopRequestTime}>
                    <TimeDuration seconds={stopRequestTime - createTime}/>
                </Time>
            </strong>
        );

        values.stopUserHtml = (
            <a href={`https://${githubHost}/${stopUser}`}>
                @{stopUser}
            </a>
        );
    }

    if (conclusionTime != null) {
        conclusionTime = Math.round(getDateTime(conclusionTime) / 1000);
        values.hasConcluded = 'yes';

        values.conclusionDurationHtml = (
            <strong>
                <Time date={conclusionTime}>
                    <TimeDuration seconds={conclusionTime - (stopRequestTime || createTime)}/>
                </Time>
            </strong>
        );
    }

    return (
        <FormattedMessage
            {...messages.conclusionMessage}
            values={values}
        />
    );
}

StopLine.propTypes = {
    githubHost: PropTypes.string.isRequired,
    stopUser: PropTypes.string,
    createTime: PropTypes.oneOfType([
        PropTypes.number,
        PropTypes.string,
        PropTypes.instanceOf(Date),
    ]).isRequired,
    conclusionTime: PropTypes.oneOfType([
        PropTypes.number,
        PropTypes.string,
        PropTypes.instanceOf(Date),
    ]),
    stopRequestTime: PropTypes.oneOfType([
        PropTypes.number,
        PropTypes.string,
        PropTypes.instanceOf(Date),
    ]),
};

StopLine.defaultProps = {
    stopUser: null,
    conclusionTime: null,
    stopRequestTime: null,
};

export default StopLine;
