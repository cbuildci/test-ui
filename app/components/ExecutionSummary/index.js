/**
 * ExecutionSummary
 */

import React from 'react';
import PropTypes from 'prop-types';
import ExecutionStatus from 'components/ExecutionStatus';
import IdentifierLine from './IdentifierLine';
import StartLine from './StartLine';
import StopLine from './StopLine';
import ErrorBoundary from 'components/ErrorBoundary';

const lineRenderError = <span><em>(Render Error)</em></span>;

function ExecutionSummary({
    githubHost,
    owner,
    repo,
    commit,
    executionNum,
    status,
    createTime,
    conclusion,
    conclusionTime,
    initiator,
    stopUser,
    stopRequestTime,
}) {
    return (
        <div className="execution-summary mt-3 mb-3">
            <div className="mt-1 mb-1 execution-status">
                <ErrorBoundary errorMessage={lineRenderError}>
                    <ExecutionStatus status={conclusion || status}/>
                </ErrorBoundary>
            </div>

            <div className="mt-1 mb-1">
                <ErrorBoundary errorMessage={lineRenderError}>
                    <IdentifierLine
                        githubHost={githubHost}
                        owner={owner}
                        repo={repo}
                        commit={commit}
                        executionNum={executionNum}
                    />
                </ErrorBoundary>
            </div>

            <div className="d-flex align-items-baseline mt-1 mb-1">
                <i className="fas fa-play-circle fa-fw text-muted mr-1 flex-shrink-0"/>
                <ErrorBoundary errorMessage={lineRenderError}>
                    <StartLine
                        createTime={createTime}
                        githubHost={githubHost}
                        owner={owner}
                        repo={repo}
                        event={initiator}
                    />
                </ErrorBoundary>
            </div>

            {(conclusionTime != null || stopRequestTime != null) && (
                <div className="d-flex align-items-baseline mt-1 mb-1">
                    <i className="fas fa-stop-circle fa-fw text-muted mr-1 flex-shrink-0"/>
                    <ErrorBoundary errorMessage={lineRenderError}>
                        <StopLine
                            githubHost={githubHost}
                            stopUser={stopUser}
                            createTime={createTime}
                            conclusionTime={conclusionTime}
                            stopRequestTime={stopRequestTime}
                        />
                    </ErrorBoundary>
                </div>
            )}
        </div>
    );
}

ExecutionSummary.propTypes = {
    githubHost: PropTypes.string.isRequired,
    owner: PropTypes.string.isRequired,
    repo: PropTypes.string.isRequired,
    commit: PropTypes.string.isRequired,
    executionNum: PropTypes.number.isRequired,
    status: PropTypes.string.isRequired,
    createTime: PropTypes.oneOfType([
        PropTypes.number,
        PropTypes.string,
        PropTypes.instanceOf(Date),
    ]).isRequired,
    conclusion: PropTypes.string,
    conclusionTime: PropTypes.oneOfType([
        PropTypes.number,
        PropTypes.string,
        PropTypes.instanceOf(Date),
    ]),
    initiator: PropTypes.object,
    stopUser: PropTypes.string,
    stopRequestTime: PropTypes.oneOfType([
        PropTypes.number,
        PropTypes.string,
        PropTypes.instanceOf(Date),
    ]),
};

ExecutionSummary.defaultProps = {
    initiator: null,
    conclusion: null,
    conclusionTime: null,
    stopUser: null,
    stopRequestTime: null,
};

export default ExecutionSummary;
