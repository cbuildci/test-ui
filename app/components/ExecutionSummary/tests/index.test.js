import React from 'react';
import { snapshots } from '../../../../internals/testing/snapshot-util';
import ExecutionSummary from '../index';

describe('<ExecutionSummary />', () => {
    it('Should render expected JSX for initial props (before the step function starts)', () => {
        snapshots(
            <ExecutionSummary
                githubHost="git.com"
                owner="foo"
                repo="bar"
                commit="edd98b1508f303eabaa12464f1dbb8ec5aace9de"
                executionNum={5}
                status="QUEUED"
                createTime="2018-07-27T16:52:18.099Z"
                OFFconclusion=""
                OFFconclusionTime=""
                OFFinitiator=""
                OFFstopUser=""
                OFFstopRequestTime=""
            />
        );
    });

    it('Should render expected JSX for minimal in-progress props (after the step function starts)', () => {
        snapshots(
            <ExecutionSummary
                githubHost="git.com"
                owner="foo"
                repo="bar"
                commit="edd98b1508f303eabaa12464f1dbb8ec5aace9de"
                executionNum={5}
                status="IN_PROGRESS"
                createTime="2018-07-27T16:52:18.099Z"
                initiator={{
                    event: 'pull_request',
                    action: 'opened',
                    pull_request: {
                        head: {
                            sha: '10145a195894f17110b6d9edb9ff76697cc8d327',
                            ref: 'foobar-branch-b',
                        },
                        number: 3,
                        id: 204485519,
                        base: {
                            sha: '33bf84a4563dcd16e027c0c6cf2a08984740d3f5',
                            ref: 'foobar-branch-a',
                        },
                    },
                }}
            />
        );
    });

    it('Should render expected JSX for succeeded execution', () => {
        snapshots(
            <ExecutionSummary
                githubHost="git.com"
                owner="foo"
                repo="bar"
                commit="edd98b1508f303eabaa12464f1dbb8ec5aace9de"
                executionNum={5}
                status="COMPLETED"
                createTime="2018-07-27T16:52:18.099Z"
                initiator={{
                    event: 'pull_request',
                    action: 'opened',
                    pull_request: {
                        head: {
                            sha: '10145a195894f17110b6d9edb9ff76697cc8d327',
                            ref: 'foobar-branch-b',
                        },
                        number: 3,
                        id: 204485519,
                        base: {
                            sha: '33bf84a4563dcd16e027c0c6cf2a08984740d3f5',
                            ref: 'foobar-branch-a',
                        },
                    },
                }}
                conclusion="SUCCEEDED"
                conclusionTime="2018-07-27T17:00:00.000Z"
            />
        );
    });
});
