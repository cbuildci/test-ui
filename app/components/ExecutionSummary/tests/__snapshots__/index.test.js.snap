// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExecutionSummary /> Should render expected JSX for initial props (before the step function starts) 1`] = `
<div
  className="execution-summary mt-3 mb-3"
>
  <div
    className="mt-1 mb-1 execution-status"
  >
    <ErrorBoundaryWithContext
      errorMessage={
        <span>
          <em>
            (Render Error)
          </em>
        </span>
      }
    >
      <ExecutionStatus
        status="QUEUED"
      />
    </ErrorBoundaryWithContext>
  </div>
  <div
    className="mt-1 mb-1"
  >
    <ErrorBoundaryWithContext
      errorMessage={
        <span>
          <em>
            (Render Error)
          </em>
        </span>
      }
    >
      <IdentifierLine
        commit="edd98b1508f303eabaa12464f1dbb8ec5aace9de"
        executionNum={5}
        githubHost="git.com"
        owner="foo"
        repo="bar"
      />
    </ErrorBoundaryWithContext>
  </div>
  <div
    className="d-flex align-items-baseline mt-1 mb-1"
  >
    <i
      className="fas fa-play-circle fa-fw text-muted mr-1 flex-shrink-0"
    />
    <ErrorBoundaryWithContext
      errorMessage={
        <span>
          <em>
            (Render Error)
          </em>
        </span>
      }
    >
      <StartLine
        conclusionTime={null}
        createTime="2018-07-27T16:52:18.099Z"
        event={null}
        githubHost="git.com"
        owner="foo"
        repo="bar"
      />
    </ErrorBoundaryWithContext>
  </div>
</div>
`;

exports[`<ExecutionSummary /> Should render expected JSX for initial props (before the step function starts) 2`] = `
<div
  class="execution-summary mt-3 mb-3"
>
  <div
    class="mt-1 mb-1 execution-status"
  >
    <span
      style="color:#6c757d"
    >
      <i
        class="far fa-clock fa-fw mr-1"
      />
      <span>
        Queued
      </span>
    </span>
  </div>
  <div
    class="mt-1 mb-1"
  >
    <span
      class="d-inline-flex flex-wrap"
    >
      <span
        class="mr-2"
      >
        <i
          class="fab fa-github fa-fw text-muted mr-1"
        />
        <a
          href="https://git.com/foo/bar"
        >
          foo/bar
        </a>
      </span>
      <span
        class="mr-2"
      >
        <i
          class="fas fa-code-branch fa-fw text-muted mr-1"
        />
        <a
          href="https://git.com/foo/bar/commit/edd98b1508f303eabaa12464f1dbb8ec5aace9de"
        >
          edd98b1508
        </a>
      </span>
      <span
        class="mr-2"
      >
        <i
          class="fas fa-shipping-fast fa-fw text-muted mr-1"
        />
        #5
      </span>
    </span>
  </div>
  <div
    class="d-flex align-items-baseline mt-1 mb-1"
  >
    <i
      class="fas fa-play-circle fa-fw text-muted mr-1 flex-shrink-0"
    />
    <span>
      Started 
      <strong>
        <time
          class="position-relative"
          datetime="2018-07-27T16:52:18.099Z"
          id="foobar"
          tabindex="0"
        >
          <span>
            next year
          </span>
        </time>
      </strong>
    </span>
  </div>
</div>
`;

exports[`<ExecutionSummary /> Should render expected JSX for minimal in-progress props (after the step function starts) 1`] = `
<div
  className="execution-summary mt-3 mb-3"
>
  <div
    className="mt-1 mb-1 execution-status"
  >
    <ErrorBoundaryWithContext
      errorMessage={
        <span>
          <em>
            (Render Error)
          </em>
        </span>
      }
    >
      <ExecutionStatus
        status="IN_PROGRESS"
      />
    </ErrorBoundaryWithContext>
  </div>
  <div
    className="mt-1 mb-1"
  >
    <ErrorBoundaryWithContext
      errorMessage={
        <span>
          <em>
            (Render Error)
          </em>
        </span>
      }
    >
      <IdentifierLine
        commit="edd98b1508f303eabaa12464f1dbb8ec5aace9de"
        executionNum={5}
        githubHost="git.com"
        owner="foo"
        repo="bar"
      />
    </ErrorBoundaryWithContext>
  </div>
  <div
    className="d-flex align-items-baseline mt-1 mb-1"
  >
    <i
      className="fas fa-play-circle fa-fw text-muted mr-1 flex-shrink-0"
    />
    <ErrorBoundaryWithContext
      errorMessage={
        <span>
          <em>
            (Render Error)
          </em>
        </span>
      }
    >
      <StartLine
        conclusionTime={null}
        createTime="2018-07-27T16:52:18.099Z"
        event={
          Object {
            "action": "opened",
            "event": "pull_request",
            "pull_request": Object {
              "base": Object {
                "ref": "foobar-branch-a",
                "sha": "33bf84a4563dcd16e027c0c6cf2a08984740d3f5",
              },
              "head": Object {
                "ref": "foobar-branch-b",
                "sha": "10145a195894f17110b6d9edb9ff76697cc8d327",
              },
              "id": 204485519,
              "number": 3,
            },
          }
        }
        githubHost="git.com"
        owner="foo"
        repo="bar"
      />
    </ErrorBoundaryWithContext>
  </div>
</div>
`;

exports[`<ExecutionSummary /> Should render expected JSX for minimal in-progress props (after the step function starts) 2`] = `
<div
  class="execution-summary mt-3 mb-3"
>
  <div
    class="mt-1 mb-1 execution-status"
  >
    <span
      style="color:#17a2b8"
    >
      <i
        class="fas fa-hourglass-start fa-fw mr-1"
      />
      <span>
        In Progress
      </span>
    </span>
  </div>
  <div
    class="mt-1 mb-1"
  >
    <span
      class="d-inline-flex flex-wrap"
    >
      <span
        class="mr-2"
      >
        <i
          class="fab fa-github fa-fw text-muted mr-1"
        />
        <a
          href="https://git.com/foo/bar"
        >
          foo/bar
        </a>
      </span>
      <span
        class="mr-2"
      >
        <i
          class="fas fa-code-branch fa-fw text-muted mr-1"
        />
        <a
          href="https://git.com/foo/bar/commit/edd98b1508f303eabaa12464f1dbb8ec5aace9de"
        >
          edd98b1508
        </a>
      </span>
      <span
        class="mr-2"
      >
        <i
          class="fas fa-shipping-fast fa-fw text-muted mr-1"
        />
        #5
      </span>
    </span>
  </div>
  <div
    class="d-flex align-items-baseline mt-1 mb-1"
  >
    <i
      class="fas fa-play-circle fa-fw text-muted mr-1 flex-shrink-0"
    />
    <span>
      Started 
      <strong>
        <time
          class="position-relative"
          datetime="2018-07-27T16:52:18.099Z"
          id="foobar"
          tabindex="0"
        >
          <span>
            next year
          </span>
        </time>
      </strong>
       after 
      <strong>
        <span>
          opening
        </span>
      </strong>
      
        
      <a
        href="https://git.com/foo/bar/pull/3"
      >
        <span>
          pull request #3
        </span>
      </a>
    </span>
  </div>
</div>
`;

exports[`<ExecutionSummary /> Should render expected JSX for succeeded execution 1`] = `
<div
  className="execution-summary mt-3 mb-3"
>
  <div
    className="mt-1 mb-1 execution-status"
  >
    <ErrorBoundaryWithContext
      errorMessage={
        <span>
          <em>
            (Render Error)
          </em>
        </span>
      }
    >
      <ExecutionStatus
        status="SUCCEEDED"
      />
    </ErrorBoundaryWithContext>
  </div>
  <div
    className="mt-1 mb-1"
  >
    <ErrorBoundaryWithContext
      errorMessage={
        <span>
          <em>
            (Render Error)
          </em>
        </span>
      }
    >
      <IdentifierLine
        commit="edd98b1508f303eabaa12464f1dbb8ec5aace9de"
        executionNum={5}
        githubHost="git.com"
        owner="foo"
        repo="bar"
      />
    </ErrorBoundaryWithContext>
  </div>
  <div
    className="d-flex align-items-baseline mt-1 mb-1"
  >
    <i
      className="fas fa-play-circle fa-fw text-muted mr-1 flex-shrink-0"
    />
    <ErrorBoundaryWithContext
      errorMessage={
        <span>
          <em>
            (Render Error)
          </em>
        </span>
      }
    >
      <StartLine
        conclusionTime={null}
        createTime="2018-07-27T16:52:18.099Z"
        event={
          Object {
            "action": "opened",
            "event": "pull_request",
            "pull_request": Object {
              "base": Object {
                "ref": "foobar-branch-a",
                "sha": "33bf84a4563dcd16e027c0c6cf2a08984740d3f5",
              },
              "head": Object {
                "ref": "foobar-branch-b",
                "sha": "10145a195894f17110b6d9edb9ff76697cc8d327",
              },
              "id": 204485519,
              "number": 3,
            },
          }
        }
        githubHost="git.com"
        owner="foo"
        repo="bar"
      />
    </ErrorBoundaryWithContext>
  </div>
  <div
    className="d-flex align-items-baseline mt-1 mb-1"
  >
    <i
      className="fas fa-stop-circle fa-fw text-muted mr-1 flex-shrink-0"
    />
    <ErrorBoundaryWithContext
      errorMessage={
        <span>
          <em>
            (Render Error)
          </em>
        </span>
      }
    >
      <StopLine
        conclusionTime="2018-07-27T17:00:00.000Z"
        createTime="2018-07-27T16:52:18.099Z"
        githubHost="git.com"
        stopRequestTime={null}
        stopUser={null}
      />
    </ErrorBoundaryWithContext>
  </div>
</div>
`;

exports[`<ExecutionSummary /> Should render expected JSX for succeeded execution 2`] = `
<div
  class="execution-summary mt-3 mb-3"
>
  <div
    class="mt-1 mb-1 execution-status"
  >
    <span
      style="color:#28a745"
    >
      <i
        class="fas fa-check fa-fw mr-1"
      />
      <span>
        Succeeded
      </span>
    </span>
  </div>
  <div
    class="mt-1 mb-1"
  >
    <span
      class="d-inline-flex flex-wrap"
    >
      <span
        class="mr-2"
      >
        <i
          class="fab fa-github fa-fw text-muted mr-1"
        />
        <a
          href="https://git.com/foo/bar"
        >
          foo/bar
        </a>
      </span>
      <span
        class="mr-2"
      >
        <i
          class="fas fa-code-branch fa-fw text-muted mr-1"
        />
        <a
          href="https://git.com/foo/bar/commit/edd98b1508f303eabaa12464f1dbb8ec5aace9de"
        >
          edd98b1508
        </a>
      </span>
      <span
        class="mr-2"
      >
        <i
          class="fas fa-shipping-fast fa-fw text-muted mr-1"
        />
        #5
      </span>
    </span>
  </div>
  <div
    class="d-flex align-items-baseline mt-1 mb-1"
  >
    <i
      class="fas fa-play-circle fa-fw text-muted mr-1 flex-shrink-0"
    />
    <span>
      Started 
      <strong>
        <time
          class="position-relative"
          datetime="2018-07-27T16:52:18.099Z"
          id="foobar"
          tabindex="0"
        >
          <span>
            next year
          </span>
        </time>
      </strong>
       after 
      <strong>
        <span>
          opening
        </span>
      </strong>
      
        
      <a
        href="https://git.com/foo/bar/pull/3"
      >
        <span>
          pull request #3
        </span>
      </a>
    </span>
  </div>
  <div
    class="d-flex align-items-baseline mt-1 mb-1"
  >
    <i
      class="fas fa-stop-circle fa-fw text-muted mr-1 flex-shrink-0"
    />
    <span>
      Concluded after 
      <strong>
        <time
          class="position-relative"
          datetime="1970-01-18T17:45:10.800Z"
          id="foobar"
          tabindex="0"
        >
          <span>
            7.7 minutes
          </span>
        </time>
      </strong>
    </span>
  </div>
</div>
`;
